ui/components/multichain/pages/gator-permissions/components/review-gator-permission-item.tsx

```diff
@@ -164,23 +162,71 @@ export const ReviewGatorPermissionItem = ({
   };
 
   /**
-   * Returns the expiration date from the rules
+   * Returns the expiration date from the delegation context
+   * Extracts the expiry timestamp from a permissionContext.
+   * Based on the TimestampEnforcer contract encoding:
+   * - Terms are 32 bytes total
+   * - First 16 bytes: timestampAfterThreshold (uint128)
+   * - Last 16 bytes: timestampBeforeThreshold (uint128) - this is the expiry
    *
-   * @param rules - The rules to extract the expiration from
    * @returns The expiration date
    */
-  const getExpirationDate = useCallback(
-    (rules: GatorPermissionRule[]): string => {
-      if (!rules) {
+  const getExpirationDate = useCallback((): string => {
+    try {
+      const delegations = decodeDelegations(permissionContext);
```

Comment:
This looks like it might be usable in different places. Could we wrap it into a reusable function that returns expiry or 0 and move it to: `gator-permissions-utils` file.